.github/workflows/rules.yml

```diff
@@ -0,0 +1,73 @@
+name: Rules 
+
+on:
+
+  workflow_call:
+    inputs:
+        os:
+          required: true 
+          type: string 
+        python-version:
+          required: true 
+          type: string
+        environment:
+          required: true 
+          type: string 
+    secrets:
+      CODECOV_TOKEN:
+        required: true
+
+jobs:
+  build:
+    runs-on: ${{ inputs.os }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ inputs.python-version }}
+    - name: Install package
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest-cov
+        pip install pylint
+        echo "${{ inputs.environment }}"
+        if [[ "${{ inputs.environment }}" == "qibo" ]]; then
+          pip install .[tests,docs]
+          pip install git+https://github.com/qiboteam/qibojit
+          pip install tensorflow
+        fi
+    - name: Test with pylint
+      run: |
+        if [[ "${{ inputs.environment  }}" == "qibo" ]]; then
+          pylint src/**/*.py -E -f colorized
+          pylint src/**/*.py --exit-zero -f colorized
+        fi
+        if [[ "${{ inputs.environment  }}" == "qibolab" ]]; then
+          pylint src -E -d E1123,E1120
+        fi
```

Comment:
Also here you can avoid branching:
- one out of `src/**/*.py` or `src` should work for both
- `-f colorized` is fine for both
- testing both `-E` and `--exit-zero` is either fine for both or for none
- you can add a `inputs.pylint_disable` for the error codes to disable